### ISSUE TITLE
Registries for used in k/k CI should be on Kubernetes Community

### ISSUE THREAD

**Issue by justaugustus** [kubernetes/k8s.io#1458]
ref: @spiffxp at https://github.com/kubernetes/kubernetes/issues/97002#issuecomment-737435131:

> https://kubernetes.slack.com/archives/C09QZ4DQB/p1606896985218000
> 
> The project hosting the GCR repo was swept up by a security audit because it hadn't been properly accounted for. That change has been reverted. Now waiting to see affected jobs go back to green.
> 
> We should create a community-owned equivalent project, I'll open a followup issue for that

`gcr.io/k8s-authenticated-test` is the specific case here, but we should try to cover as many as we find.

/assign
/sig release
/area release-eng artifacts
cc: @kubernetes/release-engineering @kubernetes/ci-signal @thockin 

Tagging since they also expressed interest:
@thejoycekung @jeremyrickard @aojea

**Comment by spiffxp** [comment]
/milestone v1.21
This covers a subset of projects in umbrella issue https://github.com/kubernetes/k8s.io/issues/1469

**Comment by spiffxp** [comment]
Per https://github.com/kubernetes/k8s.io/issues/1522#issuecomment-911980801 we are now down to the test-case-specific registries mentioned in the description.  We need assistance from specific SIGs to move forward on deciding if the test cases are valid, need to be rewritten, or can be dropped.

**Comment by spiffxp** [comment]
@hh proposing a rewrite of tests to standup private registry in cluster

**Comment by spiffxp** [comment]
I'll punt this out of v1.23 once we're in test freeze.  We managed to complete kubernetes-e2e-test-images this cycle, but waiting on a deprecation announcement

**Comment by BenTheElder** [comment]
> gcr.io/k8s-authenticated-test is the specific case here, but we should try to cover as many as we find.

this and the other authenticated test repo are pretty much the worst, and just an overall antipattern

> gke-release: https://github.com/kubernetes/k8s.io/issues/1525

at this point this is basically only used to setup GPUs on GCE, the problem with exiting this one is thorny licensing, we're not depending on it in test cases, release artifacts, etc, just setting up GPUs on kube-up.sh

we could probably figure it out, but I'm not sure it's worth it. we're always going to be dependent on a third-party for the forseeable future on gpu setup, and we only have like one job using this.

the authenticated test tests are not great, and I'm leaning towards we should just delete the tests due to lack of response on sorting out a sustainable alternative (which would require input from the owners of the code under test)

hosting a public endpoint with hardcoded login / service account key in the test binaries was a terrible hack and not something we should replicate or continue to depend on, regardless of who owns it. if someone wanted to rebuild these tests they could create an in-cluster registry with auth or something (running on hostnet?)

see also: https://github.com/kubernetes/kubernetes/issues/97026, https://github.com/kubernetes/kubernetes/issues/113925
